components/sync15/src/clients/engine.rs

```diff
@@ -73,7 +73,16 @@ impl<'a> Driver<'a> {
             // clients collection, so we don't check for `is_tombstone`.
             // https://github.com/mozilla/application-services/issues/1801
             // tracks deleting these from the server.
-            let client: ClientRecord = payload.into_record()?;
+            if payload.is_tombstone() {
+                continue;
+            }
+
+            let client_res: Result<ClientRecord, serde_json::Error> = payload.into_record();
```

Comment:
You can probably just use a `match` here - something like:
```
let client = match payload.into_record() {
    Some(client) => client,
    Err(_) => continue,
```

(although logging that we did skip probably makes sense)

Reply:
Ah, yup, using the `match` is quite a bit cleaner for the error handling; will give that a shot.